File: components/dom_distiller/core/feedback_reporter.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/dom_distiller/core/feedback_reporter.h"

#include "base/metrics/histogram_macros.h"

namespace dom_distiller {

FeedbackReporter::FeedbackReporter() {}

FeedbackReporter::~FeedbackReporter() {}

// static
void FeedbackReporter::ReportQuality(bool good) {
  UMA_HISTOGRAM_BOOLEAN("DomDistiller.DistillationQuality", good);
}

}  // namespace dom_distiller

File: chrome/android/java/src/org/chromium/chrome/browser/feedback/FeedbackReporter.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.feedback;

/**
 * FeedbackReporter enables Chrome to send feedback to the feedback server.
 */
public interface FeedbackReporter {
    /**
     * Report feedback to the feedback server.
     *
     * @param collector the {@link FeedbackCollector} to use for extra data.
     */
    // clang-format off
    // TODO(crbug.com/781015): Clang isn't formatting this correctly.
    default void reportFeedback(FeedbackCollector collector) {}
        // clang-format on
}
